#159319949 setup database

damafeez opened the pull request:
#### What does this PR do?

- database setup and initial table creation
- install necessary modules

#### Description of Task to be completed?
- database persistence

#### How should this be manually tested?
`npm run dbSetup`
Any background context you want to provide?
n/a

What are the relevant pivotal tracker stories?
#159319949

Screenshots (if appropriate)
n/a

Questions:
None

damafeez committed:
chore(server): setup server

- setup database and create tables

[Finishes #159319949]

damafeez committed:
chore(server): setup database

- create tables
- add envronment variables
- add dbSetup script to package.json

[Finishes #159319949]

fidelisojeah commented on server/model/dbSetupQuery.js:
```diff
@@ -0,0 +1,40 @@
+const users = `
+DROP TABLE IF EXISTS users cascade;
```
So why exactly are you dropping this table first?

fidelisojeah replied:
You could:
`CREATE TABLE IF NOT EXISTS`

fidelisojeah commented on server/model/dbSetup.js:
```diff
@@ -0,0 +1,21 @@
+import { Client } from 'pg';
```
Move this file to the `migrations/` folder

fidelisojeah commented on server/model/dbSetupQuery.js:
```diff
@@ -0,0 +1,40 @@
+);`;
+
+const authentication = `
+DROP TABLE IF EXISTS authentication cascade;
```
Imagine a situation when you have a new table and you want to run migrations again, You would eventually lose al your tables using this step

fidelisojeah commented on server/model/dbSetupQuery.js:
```diff
@@ -0,0 +1,40 @@
+CREATE TABLE authentication(
+  userId INT,
+  username VARCHAR(30) NOT NULL UNIQUE,
+  password VARCHAR(40) NOT NULL,
```
you may want a much longer field size. Remember the password would ideally be hashed then stored in the dB.
use a much longer field size or maybe just use text.

fidelisojeah reviewed (changes requested):
Also, create `migrations/dropTables` which would drop tables if called.
`DROP TABLES IF EXISTS`

damafeez committed:
chore(server): setup server

- implement all table delete upon request
- remove table delete on initial dbSetup

[Finishes #159319949]

fidelisojeah commented:
You can merge this now
